functests/ptp/ptp.go

```
@@ -108,4 +96,437 @@ var _ = Describe("ptp", func() {
+						ptpRunnungPods = append(ptpRunnungPods, pod)
+					}
+				}
+				Expect(len(ptpRunnungPods)).To(BeNumerically(">", 0), fmt.Sprint("Fail to detect PTP slave/master pods on Cluster"))
+			})
+
+			// 25729
+			It("The interfaces support ptp can be discovered correctly", func() {
+
+				for _, pod := range ptpRunnungPods {
+					ptpSupportedInt := getPtpMasterSlaveAttachedInterfaces(pod)
+					Expect(len(ptpSupportedInt)).To(BeNumerically(">", 0), fmt.Sprint("Fail to detect PTP Supported interfaces on slave/master pods"))
+					ptpDiscoveredInterfaces := ptpDisoveredInterfaceList(nodePtpDeviceAPIPath + pod.Spec.NodeName)
+					matchCounter := len(ptpDiscoveredInterfaces)
+					Expect(matchCounter).To(BeNumerically("==", len(ptpSupportedInt)), fmt.Sprint("The interfaces discovered incorrectly"))
+					for _, inter := range ptpSupportedInt {
+						if Expect(ptpDiscoveredInterfaces).To(ContainElement(inter)) {
+							matchCounter--
+						}
+					}
+					Expect(matchCounter).To(BeNumerically("==", 0), fmt.Sprint("The interfaces discovered incorrectly"))
+				}
+			})
+
+			// 25730
+			It("The virtual interfaces should be not discovered by ptp", func() {
+
+				for _, pod := range ptpRunnungPods {
+					ptpNotSupportedInt := getNonPtpMasterSlaveAttachedInterfaces(pod)
+					ptpDiscoveredInterfaces := ptpDisoveredInterfaceList(nodePtpDeviceAPIPath + pod.Spec.NodeName)
+					for _, inter := range ptpNotSupportedInt {
+						Expect(ptpDiscoveredInterfaces).ToNot(ContainElement(inter), fmt.Sprint("The interfaces discovered incorrectly. PTP non supported Interfaces in list"))
+					}
+				}
+			})
+
+			// 25859
+			It("PTP discovery should exclude default network interface of nodes", func() {
+				collectRoutesCommand := []string{"cat", "/proc/net/route"}
+				for _, pod := range ptpRunnungPods {
+					commandOut, _ := execCommand(pod, collectRoutesCommand)
+					defaultRouteInterface, err := detectDefaultRouteInterface(commandOut)
```

Can detectDefaultRouteInterface get the pod and hide the execCommand function above?
Also, can you move it to the pod package under utils/pod as it feels like it could be used somewhere else?

---

Move out detectDefaultRouteInterface + execCommand to the pods.go file.